beacon_node/network/src/sync/block_sidecar_coupling.rs

```diff
@@ -235,30 +321,61 @@ impl<E: EthSpec> RangeBlockComponentsRequest<E> {
                     // TODO(das): on the initial version of PeerDAS the beacon chain does not check
                     // rpc custody requirements and dropping this check can allow the block to have
                     // an inconsistent DB.
-                    return Err(format!("No columns for block {block_root:?} with data"));
+
+                    // For now, we always assume that the block peer is right.
+                    // This is potentially dangerous as we can get isolated on a chain with a
+                    // malicious block peer.
+                    // TODO: fix this by checking the proposer signature before downloading columns.
+                    let responsible_peers = column_to_peer.iter().map(|c| (*c.0, *c.1)).collect();
+                    return Err(CouplingError {
+                        msg: format!("No columns for block {block_root:?} with data"),
+                        column_and_peer: Some((responsible_peers, PeerAction::LowToleranceError)),
```

Comment:
LowToleranceError is very harsh we can disconnect everyone very quickly in case of bugs